File: Challenge/Javascript/reverseNodesInKGroups.js
// https://app.codesignal.com/challenge/p6Ro6CZYq5j3vS7DY
// Singly-linked lists are already defined with this interface:
// function ListNode(x) {
//   this.value = x;
//   this.next = null;
// }
//
reverseNodesInKGroups = (l, k) => {
    i = 1
    a = []
    b = []
    while(l) {
        b.push(l.value)
        if(i++ % k == 0) {
            a = [...a, ...b.reverse()]
            b = []
        }
        l = l.next
    }
    return b[0] ? [...a, ...b] : a
}

File: Challenge/Python/reverseNodesInKGroups.py
# Singly-linked lists are already defined with this interface:
# class ListNode(object):
#   def __init__(self, x):
#     self.value = x
#     self.next = None
#
def reverseNodesInKGroups(l, k):
    i = 1
    a = []
    b = []
    while l:
        b.append(l.value)
        if not i % k:
            b.reverse()
            a = [*a, *b]
            b = []
        l = l.next
        i += 1
    return [*a, *b] if b else a
